recipe/cmake-tests/CMakeLists.txt

```diff
@@ -0,0 +1,47 @@
+cmake_minimum_required(VERSION 3.17)
+
+project(diana)
+
+option(CUDA_FINDER "Which method find_package will use: CUDA or CUDAToolkit" "CUDA")
+option(WITH_ENABLE_LANGUAGE "Use enable_language(CUDA)" OFF)
+message(STATUS "WITH_ENABLE_LANGUAGE=${WITH_ENABLE_LANGUAGE} CUDA_FINDER=${CUDA_FINDER}")
+
+set(VALID_CUDA_FINDERS "CUDA" "CUDAToolkit" "OFF")
+if(NOT ${CUDA_FINDER} IN_LIST VALID_CUDA_FINDERS)
+    message(FATAL_ERROR "CUDA_FINDER must be one of: ${VALID_CUDA_FINDERS}")
+endif()
+
+file(GLOB cpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cc")
+file(GLOB gpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cu")
+file(GLOB include_files "${CMAKE_SOURCE_DIR}/src/*.hpp")
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+if(WITH_ENABLE_LANGUAGE)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 14)
+    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+    if(CUDA_FINDER)
+        find_package("${CUDA_FINDER}" REQUIRED)
+    endif(CUDA_FINDER)
+    add_definitions("-DWITHCUFILES")
+    add_executable(diana ${cpu_source_files} ${gpu_source_files})
+
+else(WITH_ENABLE_LANGUAGE)
+    add_executable(diana ${cpu_source_files})
+
+    if("${CUDA_FINDER}" STREQUAL "CUDA")
+        find_package("${CUDA_FINDER}" REQUIRED)
+        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler ,\"-std=c++14\"")
+        add_definitions("-DWITHCUFILES")
+        CUDA_ADD_LIBRARY(diana_gpu ${gpu_source_files})
+        target_link_libraries(diana diana_gpu)
+    elseif("${CUDA_FINDER}" STREQUAL "CUDAToolkit")
+        # This example does not link to .cu libraries
+        find_package("${CUDA_FINDER}" REQUIRED)
+        target_link_libraries(diana CUDA::cudart)
```

Comment:
Could we check the result of `CUDAToolkit_NVCC_EXECUTABLE` (https://cmake.org/cmake/help/git-stage/module/FindCUDAToolkit.html#result-variables) to ensure it's pointing to the conda `nvcc` as opposed to the system install? I've seen cmake projects that integrate non-cmake libraries have to manually invoke `nvcc` in order to build things and we'd want them to similarly use the conda wrapper.

Reply:
We should check this in both cases of `enable_language(CUDA)` as if the `CUDA` language is enabled, it changes the search path that CMake uses for this module.

Reply:
Checking below.